### DISCUSSION TITLE
on click and hooks on server side components

### DISCUSSION THREAD

**Discussion by kingofday** [vercel/next.js#67992] [Help]
### Summary

i have a menu that works with click and state hooks to show collapsed menu,  i also wants it to be ssr component because its on my header component, so how can i make it works

### Additional information

```js
i have a menu that works with click and state hooks to show collapsed menu, i also wants it to be ssr component because its on my header component, so how can i make it works in next 14 with restricted ssr component, isnt next js framework must make possible way of rendering html easier rather than impossible?!
```

**Comment by romhenri** [comment]
Tip:

You can use your Header as RSC and it imports the menu as RCC.

Ever think that a RSC can call a RCC or RSC, but never a RCC call a RSC.

**Comment by icyJoseph** [comment]
You need to refactor, and perhaps use composition. We'd need to see a React Tree structure of your component to be able to tell you how to reorganize things, but it is definitely possible!
